File: LibraryApp/Models/Entities/AppUser.cs
using Microsoft.AspNetCore.Identity;
using System.Collections.Generic;

namespace LibraryApp.Models.Entities
{
    public class AppUser: IdentityUser
    {
        public AppUser()
        {
            this.Books = new HashSet<Book>();
        }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string Street { get; set; }
        public int StreetNumber { get; set; }
        public string LocalNumber { get; set; }
        public string City { get; set; }
        public string ZipCode { get; set; }

        public virtual ICollection<Book> Books { get; set; }

    }
}

File: LibraryApp/ClientApp/app/models/appUser.ts
import { UserRegistration } from "./user.registration.interface";

export class AppUser implements UserRegistration {

    constructor(
        public firstName: string,
        public lastName: string,
        public email: string,
        public password: string,
        public street: string,
        public streetNumber: number,
        public localNumber: string,
        public city: string,
        public zipCode: string
    ) { };
}